modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientCache.java

```diff
@@ -989,6 +990,17 @@ private QueryCursor<Cache.Entry<K, V>> indexQuery(IndexQuery<K, V> qry) {
                             serDes.writeObject(out, range.lower());
                             serDes.writeObject(out, range.upper());
                         }
+                        else if (c instanceof InIndexQueryCriterion) {
```

Comment:
Please add support backward compatibility for the thin client protocol:
see: `ClientBitmaskFeature`, `ClientProtocolContext`.
For new compatibility tests see: to `JavaThinCompatibilityTest`